code/modules/projectiles/guns/syringe_gun.dm

```diff
@@ -95,3 +94,285 @@
+		return
+
+	// Refresh the mix in the syringe by pushing it into our internal beaker, and then pulling it out.
+	// The only time we /won't/ is when we dump out to a beaker.
+	if(chambered.BB.reagents.total_volume)
+		chambered.BB.reagents.trans_to(src, chambered.BB.reagents.total_volume)
+
+	reagents.trans_to(chambered.BB, get_units_per_shot())
+
+/obj/item/gun/syringe/rapidsyringe/process_chamber()
+
+	if(!length(syringes) || chambered?.BB)
+		return
+
+	var/obj/item/reagent_containers/syringe/S = syringes[1]
+	if(!S)
+		return
+
+	chambered.BB = new S.projectile_type(src)
+	update_loaded_syringe()
+	chambered.BB.name = S.name
+
+	syringes.Remove(S)
+	qdel(S)
+
+// Unload an empty syringe, making sure its existing contents get returned to the reservoir
+/obj/item/gun/syringe/rapidsyringe/attack_self(mob/living/user)
+	if(!length(syringes) && !chambered.BB)
+		to_chat(user, "<span class='notice'>[src] is empty.</span>")
+		return FALSE
+
+	var/obj/item/reagent_containers/syringe/S
+	if(chambered.BB) // Remove the chambered syringe first
+		S = new()
+		// Dump it into the main reservoir beforehand
+		chambered.BB.reagents.trans_to(src, chambered.BB.reagents.total_volume)
+		qdel(chambered.BB)
+		chambered.BB = null
+	else
+		S = syringes[length(syringes)]
+
+	user.put_in_hands(S)
+	syringes.Remove(S)
+	process_chamber()
+	playsound(src, "sound/weapons/gun_interactions/remove_bullet.ogg", 25, 1)
+	to_chat(user, "<span class='notice'>You unload [S] from [src].</span>")
+	return TRUE
+
+/obj/item/gun/syringe/rapidsyringe/suicide_act(mob/user)
+
+	if(!chambered?.BB)
+		visible_message("<span class='danger'>[user] puts [user.p_their()] mouth to [src]'s reagent port and swings [user.p_their()] head back, it looks like [user.p_theyre()] trying to commit suicide!</span>")
+		if(!reagents.total_volume)
+			to_chat(user, "<span class='userdanger'>...but the reservoir is empty!</span>")
+			return SHAME
+		else
+			reagents.trans_to(user, reagents.total_volume)
+	else
+		visible_message("<span class='danger'>[user] raises [src] to [user.p_their()] shoulder, it looks like [user.p_theyre()] trying to lose their medical license!</span>")
+		handle_suicide(user, user)
+	return TOXLOSS
+
+/// Version that comes pre-loaded with a given amount of syringes.
+/obj/item/gun/syringe/rapidsyringe/preloaded
+	/// The number of syringes to reload. If not set, defaults to max_syringes.
+	var/number_to_preload
+
+/obj/item/gun/syringe/rapidsyringe/preloaded/Initialize(mapload)
+	. = ..()
+	var/obj/item/reagent_containers/syringe/new_syringe
+	for(var/i in 0 to (number_to_preload == null ? max_syringes : number_to_preload))
+		new_syringe = new()
+		insert_single_syringe(new_syringe)
+
+/// Version that comes loaded with half of the standard amount of syringes. Used in the uplink.
+/obj/item/gun/syringe/rapidsyringe/preloaded/half
+
+/obj/item/gun/syringe/rapidsyringe/preloaded/half/Initialize(mapload)
+	number_to_preload = max_syringes / 2
+	. = ..()
```

Comment:
This order of operations is killing me

Reply:
yeah, since I add the new syringes in the parent initialize it's a necessary evil